docs/proofs/interactive.rst

```diff
@@ -25,145 +123,147 @@ hypothesis, and insert *holes* for the rest of the definition:
+    {hole_0} : (k : Nat) -> (k = plus k 0) -> S k = S (plus k 0)
 
 In this case, the goal is a function type, using ``k`` (the argument
 accessible by pattern matching) and ``ih`` — the local variable
 containing the result of the recursive call. We can introduce these as
-premises using the ``intro`` tactic twice (or ``intros``, which
-introduces all arguments as premises). This gives:
+assumptions using the ``intro`` tactic twice. The parameter is entered as
+a constant of type ``TTName`` which is entered as a backtick with double
+braces `{{ih}}. This gives:
 
 .. code-block:: idris
 
-      k : Nat
-      ih : k = plus k Z
-    ---------------------------------- (plusredZ_S) --------
-    {hole2} : S k = plus (S k) Z
+    -Main.plusredZ_S> intro `{{k}}
+
+    ----------              Assumptions:              ----------
+     k : Nat
+    ----------                 Goal:                  ----------
+    {hole_0} : (k = plus k 0) -> S k = S (plus k 0)
+    -Main.plusredZ_S> intro `{{ih}}
+
+    ----------              Assumptions:              ----------
+     k : Nat
+     ih : k = plus k 0
+    ----------                 Goal:                  ----------
+    {hole_0} : S k = S (plus k 0)
 
 Since plus is defined by recursion on its first argument, the term
 ``plus (S k) Z`` in the goal can be simplified, so we use ``compute``.
 
 .. code-block:: idris
 
-      k : Nat
-      ih : k = plus k Z
-    ---------------------------------- (plusredZ_S) --------
-    {hole2} : S k = S (plus k Z)
+    -Main.plusredZ_S> compute
+
+    ----------              Assumptions:              ----------
+     k : Nat
+     ih : k = plus k 0
+    ----------                 Goal:                  ----------
+    {hole_0} : S k = S (plus k 0)
+
 
 We know, from the type of ``ih``, that ``k = plus k Z``, so we would
 like to use this knowledge to replace ``plus k Z`` in the goal with
-``k``. We can achieve this with the ``rewrite`` tactic:
+``k``. We can achieve this with the ``rewriteWith`` tactic:
 
 .. code-block:: idris
 
-    -plusredZ_S> rewrite ih
+    -Main.plusredZ_S> rewriteWith (Var `{{ih}})
 
-      k : Nat
-      ih : k = plus k Z
-    ---------------------------------- (plusredZ_S) --------
-    {hole3} : S k = S k
+    ----------              Assumptions:              ----------
+     k : Nat
+     ih : k = plus k 0
+    ----------                 Goal:                  ----------
+    {hole_0} : S k = S k
 
-    -plusredZ_S>
-
-The ``rewrite`` tactic takes an equality proof as an argument, and tries
-to rewrite the goal using that proof. Here, it results in an equality
-which is trivially provable:
+The ``rewriteWith`` tactic takes an equality proof as an argument, and tries
+to rewrite the goal using that proof. The ih value is entered as a constant
+of type ``TTName`` which is entered as a backtick with double braces `{{ih}} but
+``rewriteWith`` requires a ``raw`` value so we use ``Var`` to do this conversion.
```

Comment:
`raw` should be capitalized.

I'd phrase this something like "`rewriteWith` requires an expression, rather than just a name, so the `Var` constructor is used to make a variable".